src/core-components/SwapInput/SwapInput.tsx

```diff
@@ -0,0 +1,57 @@
+import './SwapInput.less';
+
+import React from 'react';
+
+import { escapeRegExp } from '../../utils/format';
+import { Input } from '../index';
+
+const inputRegex = RegExp(`^\\d*(?:\\\\[.])?\\d*$`); // match escaped "." characters via in a non-capturing group
```

Comment:
Two more use cases:
1. When the user types `.` as a first symbol, the Components displays and uses the `0.` value.
2. When the user types `,` symbol, the Component replaces it by the `.` symbol